src/sparse/merkle_tree.rs

```diff
@@ -30,9 +30,24 @@ pub struct MerkleTree<StorageType> {
     storage: StorageType,
 }
 
+/// The table of the sparse merkle's tree nodes. [`MerkleTree`] works with it as a sparse merkle
+/// tree, where the storage key is `Bytes32` and the value is the [`Buffer`](crate::sparse::Buffer)
+/// (raw presentation of the [`Node`](crate::sparse::Node)).
+pub struct NodesTable;
+
+impl Mappable for NodesTable {
+    /// The 32 bytes unique key of the merkle node.
+    type Key = Bytes32;
+    /// The merkle node data with information to iterate over the tree.
+    // TODO: Use directly [`Node`](crate::sparse::Node) because it will not add overhead.
```

Comment:
It is more about consistent usage of `Node` in the binary merkle tree, sparse merkle tree, and binary merkle sum tree=)

Usage of `Buffer` in one of them and `Node` in two others look not logical. To use `Buffer`, the developer still needs to convert it into `Node`(and do the reverse operation to put it into the storage).

The `sparse::Node` can store it as a raw array under the hood. We don't need to use `serde`, the binary format is suitable=) The problem can be that `Node` doesn't implement serialization and deserialization(trait) required by Postgres or MySql. But it can be solved in several ways:
- On the storage side, `Node` can be wrapped into a new struct, and traits can be implemented for this struct.
- We can implement those traits in `fuel-merkle`, but we need to know which traits to implement(it breaks abstraction).
- We can implement `AsRef<[u8]>`, `From<Buffer>`, `From<Node>` traits, and on the storage side, we still will work with slice/array and can use binary format. (I like that approach, but we will have a problem during deserialization because we need to use `Buffer` and after `into`)

In this case, we don't have any performance impact. And code looks consistent, and it is easy to use.

Reply:
Consistency is good! I agree that we should have a consistent pattern across all Merkle tree storage types. Actually, I was considering refactoring the Binary and Binary Sum tree to use a similar pattern as the Sparse tree, but I didn't come to a conclusion on that yet. This discussion is all part of that decision!

On reason I wanted to hide the use of `Node` (Sparse, Binary, or Binary Sum) from the `Storage` backings is that `Node` is an internal concept. It should only be used inside the tree implementation itself. None of the public methods on the trees actually take in or return a `Node`: they all operate on binary arrays (`&[u8]` or `Bytes32`). `Buffer` is simply a type alias for `[u8; 69]`. 

But as you suggest, we can still implement traits on top of a publicly exposed struct. We can even call that public struct `Node` too, since that is intuitive. 

Ultimately, I agree with the goals you are trying to achieve, and I think we should investigate further how to achieve them. At this moment, I think further discussion is outside the scope of this PR and I don't want to block it. We can continue this discussion at a later point!

Reply:
@bvrooman could you summarize this into a backlog issue so we don't lose track of this concern?

Reply:
Created an issue here: https://github.com/FuelLabs/fuel-merkle/issues/113